### PR TITLE
fix seq_i, seq_j type

### PR DESCRIPTION
**Pull request by amnona**
Fix the sub_seq_j = seq_i.np_sequence[:length] typo bug to seq_j.

Also, fix the excessive '-' at end of alignment being counted as mismatch when handling the indel.

Also fix the unit test, and validate the indel removal using the indel error rate and not the general mismatch error rate (by adding an indel containing sequence that should not be removed under the indel error rate, but will be removed with the mismatch error rate).

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # deblur changelog
 
+## Version 1.0.5-dev
+
+### Bug fixes
+* Fixed problem causing deblur to ignore the indel error uper bound and use the mismatch error upper bound instead. This caused deblur to use default error rate of 0.06/0.02 instead of 0.01 which is the default for indel.
```
Comment (wasade) [inline ChangeLog.md]: uper -> upper

Would it be possible to include a hyperlink to the issue? e.g., "See [issue #178](https://github.com/biocore/deblur/issues/178) for more detail"

File: ChangeLog.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # deblur changelog
 
+## Version 1.0.5-dev
```
Comment (wasade) [inline ChangeLog.md]: I don't understand why the version number has been incremented, this should still fall under 1.0.4-dev right?

**Comment by amnona** [comment]
Fixed comments
also, changed the default indel upper bound to 0.02 (instead of 0.01). This should make results more compatible with the previous version (since it used the mismatch bound which is 0.02 for 2 mismatches, which is what we get if the indel is not recognized).

File: ChangeLog.md [hunk 3]
```diff
@@ -3,6 +3,8 @@
 ## Version 1.0.4-dev
 
 ### Bug fixes
+* Fixed problem causing deblur to ignore the indel error uper bound and use the mismatch error upper bound instead. This caused deblur to use the mismatch depenent default error rate instead of the default indel error rate (which is constant for up to 3 indels). Also increased the default indel error rate to 0.02. See [issue #178](https://github.com/biocore/deblur/issues/178) for more details.
```
Comment (wasade) [inline ChangeLog.md]: Is there empirical data to support the change?

**Comment by amnona** [comment]
Since all deblur testing and usage was done with the bug, updating the
indel probability to 0.02 makes the results more compatible.
Also, 0.02 is a more conservative upper bound.

**Comment by antgonza** [comment]
after discussing with @wasade we are gonna merge and then fix in another pr ...

**Pull request merged**